src/features/post/postSlice.ts

```diff
@@ -225,6 +225,12 @@ export const savePost =
   async (dispatch: AppDispatch, getState: () => RootState) => {
     const oldSaved = getState().post.postSavedById[postId];
 
+    const { upvoteOnSave } = getState().settings.general.posts;
+
+    if (upvoteOnSave) {
```

Comment:
```suggestion
    if (upvoteOnSave && save) {
```

(If the user unsaves the post, let's not unvote it.)